streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -474,6 +475,10 @@ public static StreamThread create(final TopologyMetadata topologyMetadata,
         taskManager.setMainConsumer(mainConsumer);
         referenceContainer.mainConsumer = mainConsumer;
 
+        final String stateUpdaterId = stateUpdaterEnabled ? threadId.replace("-StreamThread-", "-StateUpdater-") : "NA";
```

Comment:
This "NA" seems a little inelegant. Philosophically, is there a special state updater with ID "NA", or is there no state updater? I think the latter, in which case I would prefer not to have a fake ID at all.

Reply:
Fair enough, reflecting some, I'm unsure what I was getting at.  Updated

Reply:
Well, I really meant don't pass in a state updater ID at all if there is no state updater enabled, but it's a very small point.

Reply:
At one point, the state updater was experimental and controlled by an internal config.  I did some follow-up, and as of the 3.8 release, it's enabled so I think we are good here.

Reply:
Thanks for looking into it.